netket/operator/_kinetic.py

```diff
@@ -61,17 +86,12 @@ def _expect_kernel(
         def logpsi_x(x):
             return logpsi(params, x)
 
-        dlogpsi_x = jax.grad(logpsi_x)
-
-        dp_dx, f_jvp = jax.linearize(dlogpsi_x, x)
-        dp_dx = dp_dx**2
-
-        basis = jnp.eye(x.shape[0], dtype=dp_dx.dtype)
+        dlogpsi_x = jacrev(logpsi_x)
 
-        dp_dx2 = jnp.diag(jax.vmap(f_jvp)(basis))
+        dp_dx2 = jnp.diag(jacfwd(dlogpsi_x)(x)[0].reshape(x.shape[0], x.shape[0]))
+        dp_dx = dlogpsi_x(x)[0][0] ** 2
```

Comment:
it's not entirely clear to me this is the correct expression for the complex case (I would have expected a modulus square, but I guess I am wrong?)

Reply:
also, there might be a way to do less effort and compute only the real part of the kinetic energy estimator, maybe directly with jax jvp without using our custom version?

Reply:
So the momentum operator squared is proportional to nabla^2 and since this nabla is w.r.t. the coordinates (which are real) I don't see where a modulus comes from.. but I might also be wrong.